Bottom: d7556f180e818d43dcc7e7df7d156547806f2356
Top:    c3b21405946ba8ed114c5478b5e605264088db09
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-13 01:54:00 +0100

gtk/gl: Don't set "magenta" for a NULL color passed to ghid_set_color

The code has a strdup (name), followed by a test for name == NULL. If
name _were_ ever NULL, the strdup would crash, so we can remove this
test.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index ad5ff98..3902bc3 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -369,14 +369,6 @@ ghid_set_color (hidGC gc, const char *name)
     }
 
   old_name = strdup (name);
-
-  if (name == NULL)
-    {
-      fprintf (stderr, "%s():  name = NULL, setting to magenta\n",
-               __FUNCTION__);
-      name = "magenta";
-    }
-
   gc->colorname = (char *) name;
 
   if (gport->colormap == NULL)
